salt/utils/vmware.py

```diff
@@ -187,71 +303,61 @@ def get_service_instance(host, username, password, protocol=None, port=None):
     service_instance = GetSi()
     if service_instance:
         if service_instance._GetStub().host == ':'.join([host, str(port)]):
-            return service_instance
+            log.trace('Using cached service instance')
+            if salt.utils.is_proxy():
```

Comment:
I see why now, it took a little while to track that down.  I think we can merge this.